templates/opps-list-widget.php

```diff
@@ -0,0 +1,39 @@
+<?php
+/**
+ * This template is used to display the list of flexible volunteer opportunities
+ * for the widget.
+ *
+ * To adjust this template copy it into your current theme within a folder called "wivm".
+ */
+
+// Global variables that are set in widget/class-widget.php via the widget settings in admin 
```

Comment:
@jg314

**Pull request updated**:

I was able to modify the plugin/widget functionality so that there is no need to use a global variable! I've commented the changes extensively in the code and here is a recap:

I added a load_template function to WI_Volunteer_Management_Template_Loader. It is a slight modification of the core WordPress load_template function which accepts an additional array parameter which makes variables from the script that called the get_template_part function of the WI_Volunteer_Management_Template_Loader class available in the template file that is being required. Hence, no longer needing to use a global variable! Instead, I just pass the $wivm_widget_options array to the get_template_part function which continues to pass it through to the newly defined load_template function making it available to required template.

I've tested this and all plugin templates are still working properly. I've also tested this with templates that overwrite our plugin templates (by placing them in a "wivm" folder inside the current theme) and all is good.